lib/storage/client.js

```diff
@@ -1106,6 +1107,69 @@ class Client extends CryptoConsumer {
     let group = Group.decode(groupJson)
     return group
   }
+
+  /**
+   * ReadGroup makes call to TozStore to read group by groupId (uuid).
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   *
+   * @returns {Group} A group from TozStore.
+   */
+  async readGroup(groupID) {
+    let request = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupID}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let storedGroup = await checkStatus(request)
+    let groupJson = await storedGroup.json()
+    let group = Group.decode(groupJson)
+    return group
+  }
+
+  /**
+   *
+   * @param {string} clientID   UUID assigned by TozStore, used to identify the client; optional.
+   * @param {Array} groupNames  Array of string group names to filter responses by; optional.
+   * @param {number} nextToken  Indicates where to start pagination; optional.
+   * @param {number} max        The maximum number of groups to list per request; optional.
+   *
+   * @return {Array}  An array of groups returned from TozStore.
+   */
+  async listGroups(
+    clientID = null,
+    groupNames = [],
+    nextToken = null,
+    max = null
+  ) {
+    const urlData = {
+      client_id: clientID,
+      group_names: groupNames,
+      nextToken: nextToken,
+      max: max,
+    }
+    let encodedUrl = urlEncodeData(urlData)
```

Comment:
huh... I think this helper was actually written to url-encode POST data in request bodies, but this is a great use for it. Nice!